renter/meta.go

```diff
@@ -200,7 +202,11 @@ func WriteMetaFile(filename string, m *MetaFile) error {
 	if err != nil {
 		return errors.Wrap(err, "could not create archive")
 	}
-	defer f.Close()
+	defer func() {
+		if e := f.Close(); e != nil && !errors.Is(e, os.ErrClosed) {
+			err = multierror.Append(err, e)
+		}
+	}()
```

Comment:
Actually, it should remove the temporary file here if exists. If writing a metafile fails, no one removes it.

Reply:
Failure to `Close` does not indicate that the file is corrupt, so removing it would be premature. This behavior is in line with standard library functions such as `ioutil.WriteFile`.